// fb-messenger-bot-api/dist/classes/buttons/URLButton.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Button_1 = require("./Button");
/**
 * Represents a URL Button. Used to represent shape of Facebook API object. Type returned by builder and exposed to users to create
 * own object of proper shape if desired taking into account content restrictions for the object.
 *
 * Check link for content restrictions: https://developers.facebook.com/docs/messenger-platform/reference/buttons/url
 */
class URLButton extends Button_1.Button {
    constructor(builder) {
        super(builder.getType());
        this._url = builder.getUrl();
        this._title = builder.getTitle();
        this._messenger_extensions = builder.getMessengerExtensions();
        this._fallback_url = builder.getFallbackUrl();
        this._webview_share_button = builder.getWebviewShareButton();
        this._webview_height_ratio = builder.getWebviewHeightRatio();
    }
    get url() {
        return this._url;
    }
    get title() {
        return this._title;
    }
    get messenger_extensions() {
        return this._messenger_extensions;
    }
    get fallback_url() {
        return this._fallback_url;
    }
    get webview_height_ratio() {
        return this._webview_height_ratio;
    }
    get webview_share_button() {
        return this._webview_share_button;
    }
}
exports.URLButton = URLButton;
